tests/encryption_/test_base.py

```diff
@@ -0,0 +1,224 @@
+            patient_name="John Doe",
+            patient_notes="patient notes " * 25,
+            registration_date=datetime(2023, 10, 1, 12, 0, 0),
+            is_active=True,
+            email="john.doe@example.com",
+        )
+
+        # TODO: Embed billing and patient_record models in patient model
+        # then add tests
+
+    def test_get_encrypted_fields_map(self):
+        """Test class method called by schema editor
+        and management command to get encrypted fields map for
+        `create_encrypted_collection` and `auto_encryption_opts` respectively.
+        There are no data keys in the results.
+
+        Data keys for the schema editor are created by
+        `create_encrypted_collection` and data keys for the
+        management command are created by the management command
+        using code similar to the code in `create_encrypted_collection`
+        in Pymongo.
+        """
+        expected_encrypted_fields_map = {
+            "fields": [
+                {
+                    "bsonType": "int",
+                    "path": "patient_id",
+                    "queries": {"queryType": "equality"},
+                },
+                {
+                    "bsonType": "string",
+                    "path": "patient_name",
+                },
+                {
+                    "bsonType": "string",
+                    "path": "patient_notes",
+                    "queries": {"queryType": "equality"},
+                },
+                {
+                    "bsonType": "date",
+                    "path": "registration_date",
+                    "queries": {"queryType": "equality"},
+                },
+                {
+                    "bsonType": "bool",
+                    "path": "is_active",
+                    "queries": {"queryType": "equality"},
+                },
+                {
+                    "bsonType": "string",
+                    "path": "email",
+                    "queries": {"queryType": "equality"},
+                },
+            ]
+        }
+        self.maxDiff = None
+        connection = connections["encrypted"]
+        auto_encryption_opts = getattr(connection.connection._options, "auto_encryption_opts", None)
+        with connection.schema_editor() as editor:
+            client = connection.connection
+            encrypted_fields_map = editor._get_encrypted_fields_map(
+                self.patient, client, auto_encryption_opts
+            )
+            for field in encrypted_fields_map["fields"]:
+                # Remove data keys from the output
+                field.pop("keyId", None)
+            self.assertEqual(
+                encrypted_fields_map,
+                expected_encrypted_fields_map,
+            )
+
+    def test_set_encrypted_fields_map_in_client(self):
+        # TODO: Create new client with and without schema map provided then
+        # sync database to ensure encrypted collections are created in both
```

Comment:
Is this a blocker for this PR?

Reply:
No, this is currently done manually and having a unit test is not a blocker for the PR. I'll update the comment to describe the manual process.